mentoring/answer.py

```diff
@@ -114,31 +214,65 @@ def save(self):
         if self.name:
             answer_data = self.get_model_object()
-            if answer_data.student_input != self.student_input and not self.read_only:
+            if answer_data.student_input != self.student_input:
                 answer_data.student_input = self.student_input
                 answer_data.save()
 
-    def get_model_object(self, name=None):
+    @classmethod
+    def get_template(cls, template_id):
         """
-        Fetches the Answer model object for the answer named `name`
+        Used to interact with Studio's create_xblock method to instantiate pre-defined templates.
         """
-        # By default, get the model object for the current answer's name
-        if not name:
-            name = self.name
-        # Consistency check - we should have a name by now
-        if not name:
-            raise ValueError, 'AnswerBlock.name field need to be set to a non-null/empty value'
+        # Generate a random 'name' value
+        if template_id == 'studio_default':
+            return {'data': {'name': uuid.uuid4().hex[:7]}}
+        return {'metadata': {}, 'data': {}}
 
-        # TODO: Why do we need to use `xmodule_runtime` and not `runtime`?
-        student_id = self.xmodule_runtime.anonymous_student_id
-        course_id = self.xmodule_runtime.course_id
 
-        answer_data, created = Answer.objects.get_or_create(
-            student_id=student_id,
-            course_id=course_id,
-            name=name,
-        )
-        return answer_data
+@XBlock.needs("i18n")
+class AnswerRecapBlock(AnswerMixin, StudioEditableXBlockMixin, XBlock):
+    """
+    A block that displays an answer previously entered by the student (read-only).
+    """
+    name = String(
+        display_name=_("Question ID"),
+        help=_("The ID of the question for which to display the student's answer."),
+        scope=Scope.content,
+    )
+    display_name = String(
+        display_name=_("Title"),
+        help=_("Title of this answer recap section"),
+        scope=Scope.content,
+        default="",
+    )
+    description = String(
+        display_name=_("Description"),
+        help=_("Description of this answer (optional). Can include HTML."),
+        scope=Scope.content,
+        default="",
+    )
+    editable_fields = ('name', 'display_name', 'description')
+
+    @property
+    def student_input(self):
+        if self.name:
+            return self.get_model_object().student_input
+        return ''
+
+    def fallback_view(self, view_name, context=None):
+        context = context or {}
+        context['title'] = self.display_name
+        context['description'] = self.description
+        context['student_input'] = self.student_input
+        html = loader.render_template('templates/html/answer_read_only.html', context)
+
+        fragment = Fragment(html)
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/answer.css'))
+        return fragment
```

Comment:
I reiterate my concern about using `fallback_view`, and not knowing which fields will be available.